Assets/Scripts/Score/ScoreManager.cs

```diff
@@ -0,0 +1,71 @@
+using UnityEngine;
+using UnityEngine.SceneManagement;
+using TMPro;
+
+public class ScoreManager : MonoBehaviour
+{
+    public static ScoreManager Instance { get; private set; }
+    public float Score
+    {
+        get { return score; }
+        private set { score = value; }
+    }
+
+    [SerializeField]
+    private TextMeshProUGUI scoreText;
+
+    private const float moveMulti4Pts = 10.0f, // The value to multiply the distance traveled by
+                        enemyFelledPts = 100.0f; // The points killing an enemy is worth
+
+    private float score,
+                  displayScore,
+                  pastScore,
+                  startY,
+                  transitionSpeed = 100.0f;
+
+    public void SetStartY(float y) { startY = y; }
+    
+    // Pass a positive for increase, negative for decrease
+    public void ChangeScore(float amount) { Score += amount; }
+
+    public void UpdateScoreDisplay()
+    {
+        scoreText.text = string.Format("Score: {0:00000}", displayScore);
+    }
+
+    public void EnemyFelled()
+    {
+        ChangeScore(enemyFelledPts);
+    }
+
+    public void BossFelled(float pts)
+    {
+        ChangeScore(pts);
+    }
+
+    public void AbruptLevelEnd()
```

Comment:
This method can be removed once `ScoreManager` has a reference to the player's transform that is set in the Inspector.